packages/material-ui/src/ButtonGroup/ButtonGroup.test.js

```diff
@@ -0,0 +1,229 @@
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedOutlinedSecondary), true);
+  });
+
+  it('should render a contained button', () => {
+    const wrapper = shallow(
+      <ButtonGroup variant="contained">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().variant, 'contained');
+    assert.strictEqual(wrapper.childAt(0).props().color, 'default');
+    assert.strictEqual(wrapper.hasClass(classes.contained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.grouped), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedPrimary), false);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedSecondary), false);
+  });
+
+  it('should render a contained primary button', () => {
+    const wrapper = shallow(
+      <ButtonGroup variant="contained" color="primary">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().variant, 'contained');
+    assert.strictEqual(wrapper.childAt(0).props().color, 'primary');
+    assert.strictEqual(wrapper.hasClass(classes.contained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.grouped), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedPrimary), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedSecondary), false);
+  });
+
+  it('should render a contained secondary button', () => {
+    const wrapper = shallow(
+      <ButtonGroup variant="contained" color="secondary">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().variant, 'contained');
+    assert.strictEqual(wrapper.childAt(0).props().color, 'secondary');
+    assert.strictEqual(wrapper.hasClass(classes.contained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.grouped), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedPrimary), false);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedSecondary), true);
+  });
+
+  it('should render a small button', () => {
+    const wrapper = shallow(
+      <ButtonGroup size="small">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().size, 'small');
+  });
+
+  it('should render a large button', () => {
+    const wrapper = shallow(
+      <ButtonGroup size="large">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().size, 'large');
```

Comment:
```suggestion
    assert.strictEqual(wrapper.find('button').hasClass(classes.sizeLarge), true);
```

we don't actually care about the component tree.